### DISCUSSION TITLE
Best method to limit a StrongSwan instance to a single remote host

### DISCUSSION THREAD

**Discussion by grahamqs** [strongswan/strongswan#1651] [General]
Hi

I'm looking to limit my StrongSwan gateways to only allow a single host to connect. Essentially I have a requirement for 1 client, must have it's own gateway.

Looking at the up/down script this is based on the IPsec SA coming up, which means that another client could perform an IKE exchange.

Is there any way to restict a gateway from only allowing one host ?

I've also thought about restricting the gatway using the IKE Identity, but this isn't possible, but also would allow a client to connect using IKEv2 prior to IKE_AUTH.

many thanks

**Comment by tobiasbrunner** [comment]
`remote_addrs`?

**Reply by grahamqs** [reply]
Hi Tobias

Sorry for the late reply. These are clients that are roaming, so their IP will change.

I was thinking about something in the log, as this should only see a single IKE_SA_INIT event, but wasn't sure if you had any other ideas ?

cheers

**Reply by tobiasbrunner** [reply]
> These are clients that are roaming, so their IP will change.

I don't understand. What exactly do you want to limit then? That only a single host can connect from any arbitrary IP address? Why? And why not do that based on the identity then?
